app/forms/custom_fields/hierarchy/item_form.rb

```diff
@@ -62,21 +62,19 @@ class NewItemForm < ApplicationForm
         end
       end
 
-      def initialize(custom_field:, label:, short:)
+      def initialize(custom_field:, hierarchy_item:, label:, short:)
         super()
         @custom_field = custom_field
+        @hierarchy_item = hierarchy_item
         @label = label
         @short = short
       end
 
       private
 
       def validation_message_for(attribute)
-        @custom_field
-          .errors
-          .messages_for(attribute)
-          .to_sentence
-          .presence
+        @hierarchy_item.errors.messages_for(attribute).to_sentence.presence ||
+          @custom_field.errors.messages_for(attribute).to_sentence.presence
```

Comment:
I think there's a bit of semantics here to decide if we can operate only on top of `hierarchy_item` instead of adding information to the `custom_field`. But for now I kept both.